### PR TITLE
build(snap): update kuiper to 1.0.0

### PR DESCRIPTION
**Pull request by hareshkainthcanonical**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/edgex-go/blob/master/.github/Contributing.md.

## What is the current behavior?

The current version of Kuiper is 0.4.1.

## What is the new behavior?
Update Kuiper to a newer version and align with PR https://github.com/edgexfoundry/developer-scripts/pull/326.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## New Imports

- [x] Yes
- [ ] No

Updated Kuiper from 0.4.1 to 1.0.0

## Specific Instructions
Are there any specific instructions or things that should be known prior to reviewing?
There are now two new directories under /var/snap/edgexfoundry/current/kuiper/ named log and plugins. 

## Other information
For additional information in regards to plugins see https://github.com/emqx/kuiper/blob/master/docs/en_US/plugins/overview.md

For testing I used the example from https://docs.edgexfoundry.org/1.2/examples/Ch-CommandingDeviceThroughRulesEngine/

### CODE REVIEW

File: snap/snapcraft.yaml [hunk 1]
```diff
@@ -377,14 +377,24 @@ apps:
...
-      EDGEX_PROTOCOL: tcp
-      EDGEX_PORT: 5566
+#      # KUIPER_DEBUG: "true"
+#      KUIPER_CONSOLE_LOG: "true"
+#      KUIPER_REST_PORT: 48075
+#      EDGEX_SERVER: localhost
+#      EDGEX_SERVICE_SERVER: http://localhost:48080
+#      EDGEX_TOPIC: events
+#      EDGEX_PROTOCOL: tcp
+#      EDGEX_PORT: 5566
```
Comment (lenny-intel) [inline snap/snapcraft.yaml:387]: Remove commented out settings.

**Review by lenny-intel — CHANGES_REQUESTED** [review]

File: snap/snapcraft.yaml [hunk 2]
```diff
@@ -950,7 +960,7 @@ parts:
 
   kuiper:
     source: https://github.com/emqx/kuiper.git
-    source-tag: 0.4.1
+    source-tag: 0.9.1
```
Comment (lenny-intel) [inline snap/snapcraft.yaml:963]: Please change to version 1.0.0, which is now released and what we want for code freeze. I will be updating the compose files for this version today.

**Review by lenny-intel — CHANGES_REQUESTED** [review]

File: snap/snapcraft.yaml (hunk 1 above)
Comment (hareshkainthcanonical) [reply snap/snapcraft.yaml]: Removed commented out settings.

File: snap/snapcraft.yaml (hunk 2 above)
Comment (hareshkainthcanonical) [reply snap/snapcraft.yaml]: Updated to 1.0.0

**Review by lenny-intel — APPROVED** [review]
LGTM

**Review by tonyespy — CHANGES_REQUESTED** [review]
A couple of comments...

Your commit message needs some re-work:

```
feat(snap): kuiper source tag set to 1.0.0 in snapcraft.yaml …

Updated the source-tag to 1.0.0 to align with PR edgexfoundry/developer-scripts#326.
```

  * This isn't really a new feature, so "feat" isn't appropriate. We've been using "fix(snap)" whenever we update parts, however you could also argue that "build(snap)" would be appropriate as well in this situation.
  * Commit message subject lines are supposed to use the imperative mood (i.e. fix this, merge that, update this, ...). Also you've changed more than just `snapcraft.yaml` here, so why mention it in the commit message? Since you can't list every file you change in the commit message, in general it doesn't make sense to mention file names at all, plus including it makes the subject line > 50 characters. If you look through the git history for some recent snap changes where we've updated parts, simple is better (e.g. `fix(snap): update kong to 2.0.5`). Also, I'm not sure it's even worth including a commit message body, as again you talk about the source-tag only, when there are other changes included.

In your PR description, you checked "Tests for the changes have been added (for bug fixes / features)", yet I don't see any tests being added by this PR?

And finally, in the PR description under "Other Information", it would be helpful to include a short description of how you tested this PR.

File: snap/snapcraft.yaml [hunk 3]
```diff
@@ -393,22 +393,23 @@ apps:
     plugs: [home, removable-media, network]
   kuiper:
     adapter: full
-    command: bin/kuiper-server
+    command: bin/kuiperd
```
[the remark this answers is not in the document]
Comment (hareshkainthcanonical) [reply snap/snapcraft.yaml:396]: Release page from Kuiper (https://github.com/emqx/kuiper/releases) for version 1.0.0:

> A break change! If any program or scripts that references to Kuiper binary file name bin/server & bin/cli, you'll have to update correspondingly as in below.
    Renamed bin/server to bin/kuiperd
    Renamed bin/cli to bin/kuiper"

I renamed any reference of kuiper-server to kuiperd and kuiper-cli to kuiper. I was then able to build the snap without any further issue.

File: snap/snapcraft.yaml [hunk 4]
```diff
@@ -393,22 +393,23 @@ apps:
...
+      EDGEX__DEFAULT__SERVER: localhost
+      EDGEX__DEFAULT__SERVICESERVER: http://localhost:48080
+      EDGEX__DEFAULT__TOPIC: events
+      KUIPER__BASIC__CONSOLELOG: "true"
+      KUIPER__BASIC__RESTPORT: 48075
     plugs: [network, network-bind]
   kuiper-cli:
     adapter: full
-    command: bin/kuiper-cli
+    command: bin/kuiper
```
[the remark this answers is not in the document]
Comment (hareshkainthcanonical) [reply snap/snapcraft.yaml:412]: Release page from Kuiper (https://github.com/emqx/kuiper/releases) for version 1.0.0:

> A break change! If any program or scripts that references to Kuiper binary file name bin/server & bin/cli, you'll have to update correspondingly as in below.
    Renamed bin/server to bin/kuiperd
    Renamed bin/cli to bin/kuiper"

I renamed any reference of kuiper-server to kuiperd and kuiper-cli to kuiper. I was then able to build the snap without any further issue.

File: snap/snapcraft.yaml [hunk 5]
```diff
@@ -792,12 +793,13 @@ parts:
...
+      cp -a $SNAPCRAFT_PART_SRC/plugins/. $SNAPCRAFT_PART_INSTALL/plugins
 
       install -DT "$SNAPCRAFT_PART_SRC/LICENSE" \
          "$SNAPCRAFT_PART_INSTALL/usr/share/doc/kuiper/LICENSE"
     organize:
-      bin/cli: bin/kuiper-cli
-      bin/server: bin/kuiper-server
+      bin/cli: bin/kuiper
+      bin/server: bin/kuiperd
     stage:
```
[the remark this answers is not in the document]
Comment (hareshkainthcanonical) [reply snap/snapcraft.yaml:803]: Release page from Kuiper (https://github.com/emqx/kuiper/releases) for version 1.0.0:

> A break change! If any program or scripts that references to Kuiper binary file name bin/server & bin/cli, you'll have to update correspondingly as in below.
    Renamed bin/server to bin/kuiperd
    Renamed bin/cli to bin/kuiper"

I renamed any reference of kuiper-server to kuiperd and kuiper-cli to kuiper. I was then able to build the snap without any further issue.

**Comment by jpwhitemn** [comment]
Canonical Gents - we are now using Kuiper 1.1.1.

**Comment by tonyespy** [comment]
Closing as obsolete

**Pull request closed without merging**